pages/companies/[slug]/index.tsx

```diff
@@ -0,0 +1,82 @@
+import companies from 'api/companies';
+import dynamic from 'next/dynamic';
+import { QueryClient } from 'react-query';
+import { getQueries } from 'utils/ssr';
+import { ISlug } from 'types/slug';
+const SlugContainer = dynamic(() => import('components/slug/Container'));
+const CompanyHeader = dynamic(() => import('components/slug/CompanyHeader'));
+const CompanyProfile = dynamic(() => import('components/slug/CompanyProfile'));
+const CompanyContacts = dynamic(
+  () => import('components/slug/CompanyContacts')
+);
+const QuantityAdminsPartners = dynamic(
+  () => import('components/slug/QuantityAdminsPartners')
+);
+const TableRender = dynamic(() => import('components/slug/TableRender'));
+const PreviousInfo = dynamic(
+  () => import('components/slug/CompanyPreviousInfo')
+);
+const CompanyInvestCapital = dynamic(
+  () => import('components/slug/CompanyInvestCapital')
+);
+const SimilarCompanies = dynamic(
+  () => import('components/slug/SimilarCompanies')
+);
+const CompanyTurnover = dynamic(
+  () => import('components/slug/CompanyTurnover')
+);
+const Loading = dynamic(() => import('components/Loading'));
+
+function CompanyId({ data }: ISlug) {
+  if (!data) {
+    return <Loading />;
+  }
+  console.log(data);
+  return (
+    <div className="bg-[#fafafa]">
+      <CompanyHeader data={data} />
+      <SlugContainer>
+        <PreviousInfo data={data} />
+        <CompanyProfile data={data} />
+        <CompanyContacts data={data} />
+        <QuantityAdminsPartners data={data} />
+        <TableRender data={data} />
+        <CompanyTurnover data={data} />
+        <CompanyInvestCapital data={data} />
+        <SimilarCompanies data={data} />
+      </SlugContainer>
+    </div>
+  );
+}
+
+export default CompanyId;
+
+// CompanyId.getInitialProps = async ({ query }) => {
+//   const result = await axios.get(
+//     `/company?slug=${query.slug}`
+//   );
+
+//   return {
+//     data: result.data,
+//   };
+// };
+
+CompanyId.getInitialProps = async ({ query }): Promise<Record<string, any>> => {
```

Comment:
You did a good job! As it turned out, after understanding the documentation, I realized that I was using the query client incorrectly, so thank you for that, let's fix it as writing in documentation about SSR:
- Remove `getQueries` function in `utils/ssr.ts` and move default config to const
- For first we need to replace getStaticProps with getServerSideProps
```javascript
import { defaultOptions } from 'utils/ssr.ts'

export async function getServerSideProps ({ query }) {
  const queryClient = new QueryClient(defaultOptions);

  await queryClient.prefetchQuery(['data', { id: query.slug }], companies.search)

  return {
    props: {
      dehydratedState: dehydrate(queryClient),
    },
  }
}
```

> Important to notice about `queryKey`: **['data', { id: query.slug }]**
> This key must be unique, so the request will not be executed again

After that, we can get the cache (**data**) inside any component without re-executing request and passing data through parameters.
#### Example: components/slug/CompanyHeader.tsx
```javascript
const router = useRouter()
const queryClient = useQueryClient()

const { data } = useQuery(['data', { id: router?.query.slug }]);
{/* or */}
const data = queryClient.getQueryData(['data', { id: router?.query.slug }]);
```